src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1194,7 +1185,11 @@ private static string GetInputTypeString(InputType inputType)
             string expression)
         {
             var value = viewContext.ViewData.Eval(expression);
```

Comment:
Add comment about expecting user to provide select list items in the dictionary and (definitely) not as the `Model`.